flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -88,13 +347,280 @@ def add_wind_forcing(
+            )
+            return
+
+    def _add_forcing_discharge(self, forcing: IDischarge):
+        """Add spatially constant discharge forcing to sfincs model. Use timeseries or a constant magnitude.
+
+        Parameters
+        ----------
+        timeseries : Union[str, os.PathLike], optional
+            path to file of timeseries file (.csv) which has two columns: time and discharge, by default None
+        const_discharge : float, optional
+            time-invariant discharge magnitude [m3/s], by default None
+        """
+        # TODO investigate where to include rivers from site.toml
+        if isinstance(forcing, DischargeConstant):
+            self._model.setup_discharge_forcing(
+                timeseries=None,
+                magnitude=forcing.discharge,
+            )
+        elif isinstance(forcing, DischargeSynthetic):
+            self._model.setup_discharge_forcing(
+                timeseries=forcing.file,
+                magnitude=None,
+            )
+        else:
+            self._logger.warning(
+                f"Unsupported discharge forcing type: {forcing.__class__.__name__}"
+            )
+            return
+
+    def _add_forcing_waterlevels(self, forcing: IWaterlevel):
+        if isinstance(forcing, WaterlevelSynthetic):
+            # TODO implement this
+            pass
+        elif isinstance(forcing, WaterlevelFromFile):
+            # TODO implement this
+            pass
+        elif isinstance(forcing, WaterlevelFromModel):
+            # TODO implement this
+            pass
+        else:
+            self._logger.warning(
+                f"Unsupported waterlevel forcing type: {forcing.__class__.__name__}"
+            )
+            return
+
+    ### MEASURES HELPERS ###
+    def _add_measure_floodwall(self, floodwall: FloodWall):
+        """Add floodwall to sfincs model.
+
+        Parameters
+        ----------
+        floodwall : FloodWallModel
+            floodwall information
+        """
+        # HydroMT function: get geodataframe from filename
+        polygon_file = Database().input_path.joinpath(floodwall.attrs.polygon_file)
+        gdf_floodwall = self._model.data_catalog.get_geodataframe(
+            polygon_file, geom=self._model.region, crs=self._model.crs
+        )
+
+        # Add floodwall attributes to geodataframe
+        gdf_floodwall["name"] = floodwall.attrs.name
+        if (gdf_floodwall.geometry.type == "MultiLineString").any():
+            gdf_floodwall = gdf_floodwall.explode()
+        # TODO: Choice of height data from file or uniform height and column name with height data should be adjustable in the GUI
+        try:
+            heights = [
+                float(
+                    UnitfulLength(
+                        value=float(height),
+                        units=self._site.attrs.gui.default_length_units,
+                    ).convert(UnitTypesLength("meters"))
+                )
+                for height in gdf_floodwall["z"]
```

Comment:
ideally we would let the user choose which column in the shape file contains the height or whether the user wants to define a constant value. But I did not implement anything in the GUI because the students needed it quickly and it had no priority in our release for September. So at the moment it's a "hidden" feature, which people are not aware of.

Reply:
Sounds good, we should make an issue for this to not forget it!